kube/src/lib.rs

```diff
@@ -101,23 +101,46 @@ macro_rules! cfg_client {
         )*
     }
 }
+macro_rules! cfg_config {
+    ($($item:item)*) => {
+        $(
+            #[cfg_attr(docsrs, doc(cfg(feature = "config")))]
+            #[cfg(feature = "config")]
+            $item
+        )*
+    }
+}
+
+macro_rules! cfg_error {
```

Comment:
I think we should limit `cfg_` macros to use the names of feature sets. This reads like if `error` feature is enabled.

`cfg_client_or_config`?

Reply:
I was going by the pattern used by tokio to have the macro name be the more useful one, and it internally selected on the correct feature set: https://github.com/tokio-rs/tokio/blob/ce9ca45c92b9bd4b449f94339fbe60a4ded9ab28/tokio/src/macros/cfg.rs

Reply:
Oh, I see. Then this is fine.
I still think we should split `kube::Error` into smaller ones, but that's a separate issue.

Reply:
Yeah, more focused errors along our previous discussion is still worth doing.